internal/auth/auth.go

```diff
@@ -9,21 +9,23 @@ import (
 	"fmt"
 	"net/http"
 	"net/url"
+	"os"
 	"strings"
 	"time"
 )
 
 const (
-	clientID               = "2iZo3Uczt5LFHacKdM0zzgUO2eG2uDjT"
-	deviceCodeEndpoint     = "https://auth0.auth0.com/oauth/device/code"
-	oauthTokenEndpoint     = "https://auth0.auth0.com/oauth/token"
 	audiencePath           = "/api/v2/"
 	waitThresholdInSeconds = 3
-
 	// namespace used to set/get values from the keychain
 	SecretsNamespace = "auth0-cli"
 )
 
+var clientID = getEnv("CLIENT_ID", "2iZo3Uczt5LFHacKdM0zzgUO2eG2uDjT")
```

Comment:
we can simplify with github.com/joeshaw/envdecode
and a simple config struct in place. We could override any constant if needed, and also get rid of the logic to check the ENV.

```go
var cfg struct {
    ClientID            string `env:"AUTH0_CLIENT_ID,default=2iZo3Uczt5LFHacKdM0zzgUO2eG2uDjT"`
    BaseDomain     string `env:"AUTH0_BASE_DOMAIN,default=auth0.auth0.com"`
    DeviceCodeEndpoint     string `env:"AUTH0_DEVICE_CODE_ENDPOINT,default=https://%s/oauth/device/code"`
    OAuthTokenEndpoint     string `env:"AUTH0_OAUTH_TOKEN_ENDPOINT,default=https://%s/oauth/token"`
}

if err := envdecode.StrictDecode(&cfg); err != nil {
    return err
}
```

Reply:
Good call, I thought I remembered seeing this somewhere but couldn't find it 👍